src_c/base.c

```diff
@@ -533,6 +533,54 @@ pg_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
     return 1;
 }
 
+static int
+pg_F_IntFromObj(PyObject *obj, int *val)
+{
+    if (PyFloat_Check(obj)) {
+        *val = (int)PyFloat_AS_DOUBLE(obj);
+        return 1;
+    }
+
+    *val = PyLong_AsLong(obj);
+    if (PyErr_Occurred())
+        return 0;
+    return 1;
+}
+
+static int
+pg_F_TwoIntsFromObj(PyObject *obj, int *val1, int *val2)
+{
+    PyObject *fseqence = NULL;
+    Py_ssize_t length = 0;
+
+    if (!(fseqence = PySequence_Fast(obj, "A sequence was expected")))
+        return 0;
+
+    length = PySequence_Fast_GET_SIZE(fseqence);
+
+    if (length == 2) {
+        if (!pg_F_IntFromObj(PySequence_Fast_GET_ITEM(fseqence, 0), val1) ||
+            !pg_F_IntFromObj(PySequence_Fast_GET_ITEM(fseqence, 1), val2)) {
+            Py_DECREF(fseqence);
+            return 0;
+        }
+    }
+    else if (length == 1) {
+        /* Handle case of ((x, y), ) 'sequence inside sequence' */
+        return pg_F_TwoIntsFromObj(PySequence_Fast_GET_ITEM(fseqence, 0), val1,
```

Comment:
There's a memleak on this codepath, with `fseqence` not being decref'd. Since `fseqence` has to be freed in every condition, I suggest that it be done at the end once, and other paths should `goto` here

Reply:
Thanks i fixed it and changed the names.